src/common/bls.service.ts

```diff
@@ -22,28 +22,20 @@ export class BlsService {
       return this.publicKeysCache[key];
     }
   
-    const url = `${this.url}/validators/_doc/${key}`;
-  
-    const {
-      data: {
-        _source: { publicKeys },
-      },
-    } = await this.elasticService.get(url);
+    const url = `${this.url}/validators/_search?q=_id:${key}`;
   
+    let result = await this.elasticService.get(url);
+
+    let publicKeys = result.data.hits.hits[0]._source.publicKeys;
```

Comment:
I think we should at least check that `result.data.hits.hits > 0`

Reply:
makes sense, done in both places where the search for an individual item is performed